src/helpers.js

```diff
@@ -1,113 +1,80 @@
-                    loaded: false,
-                    
-                    // Try not to include to much dirty stuff in outcome.
-                    // If it will turn out as necessary in the future, someone will have to try to implement this.
-                    children: [],
-                    paths: []
-                };
-                
-                // This is part of what allows CJS to support circular dependencies
-                CACHE[resolvedPath] = module;
-                
-                try {
-                    loader.call(commonjsGlobal, module, module.exports);
-                } catch (ex) {
-                    // rollback
-                    delete CACHE[resolvedPath];
-                    
-                    // rethrow
-                    throw ex;
-                }
-                
-                // This is part of what allows CJS to support circular dependencies
-                module.loaded = true;
-            }
-        }
-                
-        if (module)
-            return module.exports;
-            
-        // If this is a relative path, there's only one shot at trying to figure it out.
-        if (isRelative)
-            break;
-        else {
-            // Take a step back
-            let nextDir = pathModule.normalize(pathModule.join(originalModuleDir, '../')).replace(/\\\\/g, '/');
-            if (nextDir === originalModuleDir) break;
-            originalModuleDir = nextDir;
-        }
-    }
-    
-    // Try original \`require\` with transformed path
-    return realRequire(path);
-}
-
-commonjsRequire.cache = CACHE;
+export var commonjsGlobal = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : typeof self !==
+'undefined' ? self : {};
 
 export function commonjsRegister (path, loader) {
-    DYNAMIC_REQUIRE_LOADER_MAP.set(path, loader);
+	DYNAMIC_REQUIRE_LOADER_MAP.set(path, loader);
 }
 
 export function unwrapExports (x) {
 	return x && x.__esModule && Object.prototype.hasOwnProperty.call(x, 'default') ? x['default'] : x;
 }
 
-export function createCommonjsModule(fn, module) {
-	return module = { exports: {} }, fn(module, module.exports), module.exports;
-}`;
+export function createCommonjsModule (fn, module) {
+	return module = {exports: {}}, fn(module, module.exports), module.exports;
+}
 
-export const PREFIX = '\0commonjs-proxy:';
-export const EXTERNAL = '\0commonjs-external:';
-export const DYNAMIC_REGISTER_PREFIX = '\0commonjs-dynamic-register:';
-export const DYNAMIC_PACKAGES_ID = '\0commonjs-dynamic-packages';
+let pathModule;
+const DYNAMIC_REQUIRE_LOADER_MAP = new Map();
+const DYNAMIC_REQUIRE_CACHE = Object.create(null);
+const DEFAULT_PARENT_MODULE = {
+	id: '<' + 'rollup' + '>', exports: {}, parent: undefined, filename: null, loaded: false, children: [], paths: []
+};
+const CHECKED_EXTENSIONS = ['', '.js', '.json'];
+
+export function commonjsRequire (path, originalModuleDir) {
```

Comment:
I actually think that it's not a concern, as process like uglification already removed comments, compacts usage of variables etc...

Reply:
Considering that until recently, this PR would not have produced browser compatible bundles anyway, I doubt that. But the actual reason why I do not like comments is because
* they make people wary of refactoring code because either comments become wrong or obsolete or because they need to refactor the comments as well
* they reduce the pressure to write clean, self-documenting code

I know there are situations where it is not obvious why a line was added. But rather than adding a comment, the proper way to handle these situations IMO should be to write a test that turns red when the line in question is removed. The test description then not only tells you *why* this line was added but can also motivate others to find better solutions to the root problem.